src/main/java/nextstep/qna/domain/Question.java

```diff
@@ -69,24 +37,49 @@ public void addAnswer(Answer answer) {
     }
 
     public boolean isOwner(NsUser loginUser) {
-        return writer.equals(loginUser);
+        return postContent.isOwner(loginUser);
     }
 
-    public Question setDeleted(boolean deleted) {
-        this.deleted = deleted;
-        return this;
+    public void markAsDeleted(NsUser loginUser) throws CannotDeleteException {
+        validateDeletableBy(loginUser);
+        delete();
     }
 
-    public boolean isDeleted() {
-        return deleted;
+    public List<DeleteHistory> delete(NsUser loginUser) throws CannotDeleteException {
```

Comment:
```suggestion
    public void delete(NsUser loginUser) throws CannotDeleteException {
```

테스트 때문에 이 메서드 원형을 유지했다면 테스트 코드를 수정하는 것을 추천함